clients/logger/logger.go

```diff
@@ -67,6 +67,17 @@ type fileWriter struct {
 
 // NewClient creates an instance of LoggingClient
 func NewClient(owningServiceName string, isRemote bool, logTarget string, logLevel string) LoggingClient {
+	lc := NewClientStdOut(owningServiceName, isRemote, logTarget, logLevel)
+
+	if logTarget == "" {
+		lc.Error("logTarget cannot be blank, using stdout only")
+	}
+
+	return lc
+}
+
+// NewClientStdOut creates an instance of LoggingClient that expects to log to stdout and does not check logTarget
+func NewClientStdOut(owningServiceName string, isRemote bool, logTarget string, logLevel string) LoggingClient {
```

Comment:
Refactor your implementation so NewClientStdOut doesn't require a logTarget parameter.

Reply:
Disregard.

Reply:
It's not an either/or.

Your factory functions are implementation.  They can leverage a new public method (which is not part of the LoggingClient contract) on the receiver to return what you need.

Let me know if you want to pair code this.

Reply:
Please see the latest commit.

Reply:
I did.  Commented there.